packages/yswaps/scripts/libraries/solvers/multicall/CurveSpellEth.ts

```diff
@@ -88,41 +101,55 @@ export class CurveSpellEth implements Solver {
       slippagePercentage: 10 / 100,
     });
 
-    const approveCvx = (await crv.allowance(multicallSwapperAddress, zrxCvxAllowanceTarget)) < cvxBalance;
-    if (approveCvx) await cvx.approve(zrxCvxAllowanceTarget, constants.MaxUint256);
+    const approveCvx = (await cvx.allowance(multicallSwapperAddress, zrxCvxAllowanceTarget)).lt(cvxBalance);
+    if (approveCvx) {
+      console.log('[CurveSpellEth] Approving cvx');
+      const approveCvxTx = await cvx.populateTransaction.approve(zrxCvxAllowanceTarget, constants.MaxUint256);
+      await multicallSwapperSigner.sendTransaction(approveCvxTx);
+      transactions.push(approveCvxTx);
+    }
 
+    console.log('[CurveSpellEth] Executing cvx => weth via zrx');
     const cvxToWethTx: Tx = {
       to: this.zrxContractAddress,
       data: zrxCvxData,
     };
     await multicallSwapperSigner.sendTransaction(cvxToWethTx);
+    transactions.push(cvxToWethTx);
 
-    console.log('[CurveSpellEth] Convert weth to eth');
     const wethBalance = await weth.balanceOf(multicallSwapperAddress);
-    await weth.withdraw(wethBalance);
-
-    console.log('[CurveSpellEth] Convert weth to crvSpellEth');
-    await curveSwap.add_liquidity([wethBalance, 0], 0, true, this.strategyAddress, { value: wethBalance });
+    console.log('[CurveSpellEth] Total WETH balance is', utils.formatEther(wethBalance));
+
+    const approveWeth = (await weth.allowance(multicallSwapperAddress, multicallSwapperAddress)).lt(wethBalance);
```

Comment:
sending `multicallSwapperAddress` as owner and spender? Shouldnt spender be `curveSwap.address`?

Reply:
push fix